### PR TITLE
mon/OSDMonitor: Warnings for expected_num_objects

### PR DESCRIPTION
**Pull request by fullerdj**
Warn if the user tries to create a pool with wrong or missing expected_num_objects

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -10986,6 +10986,17 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       goto reply;
     }
 
+    if (expected_num_objects == 0 &&
+	cct->_conf->osd_objectstore == "filestore" &&
+	cct->_conf->filestore_merge_threshold < 0) {
+      if (pg_num >= 1024 || pg_num / osdmap.get_num_osds() >= 50) {
```
Comment (neha-ojha) [inline src/mon/OSDMonitor.cc]: Did we change the num of pgs per osd threshold from 100 to 50?

Comment (jdurgin) [inline src/mon/OSDMonitor.cc]: should check for divide by zero here - you could have no osds

Comment (fullerdj) [reply src/mon/OSDMonitor.cc]: Didn't we? I thought we did in the last conversation we had. If you think we should keep at 100, I'm ok with that.

Comment (fullerdj) [reply src/mon/OSDMonitor.cc]: Nice catch.

**Comment by fullerdj** [comment]
@jdurgin, @neha-ojha pointed out that I mentioned 100 PGs/OSD as the threshold before (and once since) writing this code. Did we settle on 50 or was it 100 the whole time?

**Comment by jdurgin** [comment]
I don't remember 50, but I'm happy either way.


Sent from Nine
________________________________

**Comment by fullerdj** [comment]
Ok, I changed it to 100.

**Comment by yuriw** [comment]
@neha-ojha this needs to be fixed/rebased https://pastebin.com/wu5aaXZz

**Comment by yuriw** [comment]
still broken https://pastebin.com/6kdiUsgU

**Comment by yuriw** [comment]
[wip-yuri-testing-2018-07-20-1607](https://trello.com/c/UYqB4xU2/350-wip-yuri-testing-2018-07-20-1607)

**Comment by fullerdj** [comment]
[dnm] added while pgcalc discussion is ongoing.

**Comment by fullerdj** [comment]
Message updated and DNM tag removed.

**Pull request merged**